src/redux/actions/locationsAction.js

```diff
@@ -23,4 +24,16 @@ export const updateSort=()=>{
     return{
         type:SORT
     }
+}
+export const fetchLocations =()=>{
+    return (dispatch) =>{
+        axios.get('https://5e9fdb2f11b078001679ceca.mockapi.io/api/locations')
+         .then(response=>{
+             const locations=response.data;
+             dispatch(GetLocations(locations));
```

Comment:
Whenever there is data coming from outside, we need to do some tasks, before sending it to the action.
- validating the structure and correctness of data, so that it does not break the views
- adapting the plain object into a datatype (typically a JS class) for better structure

The converter methods are better written in a single file, so that it'll be the only place to change, if at all there are future changes in API data format.

Reply:
```
dispatch(getLocations(convertServiceToClientLocations(response.data)))

// in locationConverter.js file
new Location({ lat: serviceModel.latitude, lng: serviceModel.longitude})
```